packages/ai-jsx/src/core/completion.tsx

```diff
@@ -40,18 +42,50 @@ export type ModelComponent<T extends ModelPropsWithChildren> = Component<T>;
  */
 export interface FunctionDefinition {
   description?: string;
-  parameters: Record<string, FunctionParameter>;
+  parameters: FunctionParameters;
+}
+
+export function getParametersSchema(parameters: FunctionParameters) {
+  if (parameters instanceof z.Schema) {
+    const jsonSchema = zodToJsonSchema(parameters);
+    if (!('type' in jsonSchema) || jsonSchema.type !== 'object') {
+      throw new Error('Function parameters must be an object');
+    }
```

Comment:
@NickHeiner is there a sub-type of `z.Schema` that ensures the top-level to be an object? I tried a few things and asked Perplexity, but I couldn't get it to work.

Reply:
I think this does what you're looking for.

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBALzgMyhEcDkCIBNMDcAUEQMYQB2AzvAIYUCeA8gEYBWApqTAMqkAWHELTgBeRADooXaLgAUCCQFcKAawoQA7hTkBKXcRgMwHOABUxk4BWQcoAHiMmIyOPWbsuvAUNoA+EnJqeAgALnNLAG9kCDCsFlooTABfYiIAenS4AHVTTXp4GH5gKjgYCBRaYAAbMkoaOAgAJnCLcRgoJQ4CIA

https://www.perplexity.ai/search/23eea325-9d11-45af-b4d2-7daa067cc7dd?s=c